### DISCUSSION TITLE
Show up and down historical data graph on status page

### DISCUSSION THREAD

**Discussion by bvenkysubbu** [upptime/upptime#461] [Ideas]
Sometimes, a simple historical up and down graph may be better indicator of status than the response time graph. Show the uptime graph similar to this.

<img width="611" alt="Screenshot 2021-10-21 at 11 42 30 AM" src="https://user-images.githubusercontent.com/10861600/138221669-cd3b442c-efb3-4d04-b837-1d31d909c466.png">

I am willing to contribute to this feature. Please guide me to the right place to start developing this feature.

**Comment by hrst** [comment]
Could not agree more! Right now, the status page is in my humble opionion very focused on the reponse time (which by the way cannot be turned off right now, even with using `showAverageResponseTime: false`). I think it would be really nice to have more information on the status page regarding the historical uptime. This could be a monthly table with uptime percentage, and/or a more visual representation using the typical traffic light colors green->yellow/orange->red.

A few more examples:

- https://support.atlassian.com/statuspage/docs/display-historical-uptime-of-components/
- https://status.wistia.com/
- https://status.mypurecloud.com/ (see section "Genesys Cloud Platform Availability")
- https://basecamp.com/uptime/bc3

**Comment by albertkun** [comment]
@bvenkysubbu Has there been any work on this? Would love to see a "down" time graph and would help to make it happen!

**Reply by bvenkysubbu** [reply]
@albertkun this idea is gaining traction now. I haven't done any work on it so far. We can collaborate to make this happen.

**Comment by AnandChowdhary** [comment]
Would love to see this! You can add it in the upptime/status-page repository :)

**Reply by albertkun** [reply]
Sure!! @AnandChowdhary Do you (or anyone working on the upptime/status-page) know what is the best way to access the data in the `history` folder? 

Right now i'm just appending data with a curl (`curl https://api.github.com/repos/upptime/upptime/commits?path=/history/google.yml | jq . > ./data/google-uptime-history.json` but I'd rather just not write it to a file if possible!

**Reply by AnandChowdhary** [reply]
We use the GitHub API on the frontend to access the data in the same way that you're proposing, for example we already do this here: https://github.com/upptime/status-page/blob/master/src/components/Graph.svelte#L25

**Reply by albertkun** [reply]
Unfortunately, i haven't used sveldt before and am having trouble getting the server to start ( i did move the `.upptimerc.yml` to parent, but am I missing another config file or anything?

**Reply by AnandChowdhary** [reply]
No, that should be all you need. The .upptimerc.yml should be in the parent folder of where you have the status-page site if I'm not wrong :)
